tinylfu.go

```diff
@@ -15,12 +15,19 @@ type T[K comparable, V any] struct {
 	slru    *slruCache[K, V]
 	data    map[K]*list.Element[*slruItem[K, V]]
 	hash    func(K) uint64
+	drop    func(K, V)
 }
 
-func New[K comparable, V any](size int, samples int, hash func(K) uint64) *T[K, V] {
-
+func New[K comparable, V any](size int, samples int, hash func(K) uint64, drop func(K, V)) *T[K, V] {
 	const lruPct = 1
 
+	if hash == nil {
+		panic("tinylfu: hash function is required")
+	}
+	if drop == nil {
+		panic("tinylfu: drop function is required")
+	}
```

Comment:
IMHO `hash == nil` should panic (or return error) but `drop == nil` is a fine default signal to the constructor, which can install a no-op function in that case.

Reply:
That was my first intuition as well, but then I felt like it was making the code too smart; it's easy enough to declare a no-op drop function, so I opted for consistency to have all the arguments be required.